src/esthri/src/compression.rs

```diff
@@ -89,10 +99,40 @@ pub(crate) async fn compress_and_replace(path: impl AsRef<Path>) -> Result<PathB
             path.to_path_buf().with_extension(ext)
         })
         .unwrap_or_else(|| path.to_path_buf().with_extension("gz"));
-    debug!("renaming {} to {}", path.display(), file_gz.display());
-    fs::rename(temp_path, &file_gz)?;
-    let permissions = path.metadata()?.permissions();
-    fs::set_permissions(&file_gz, permissions)?;
-    fs::remove_file(path)?;
-    Ok(file_gz)
+    file_gz
+}
+
+pub(crate) fn compressed_path_to_path(path: &Path) -> PathBuf {
+    let mut pathbuf = path.to_path_buf();
+    let ext = path.extension();
+
+    if let Some(ext) = ext {
+        if ext == "gz" {
+            let filename = path.file_name().unwrap();
```

Comment:
Should probably use expects here or just some nested if lets.

Reply:
Have fixed this up now to return a result, which gets rid of some of these nastier `unwraps` :slightly_smiling_face: